### PR TITLE
get_package_deps

### PR DESCRIPTION
**Pull request by jimbrig**
for issue #129

### CODE REVIEW

**Comment by jimbrig** [comment]
@merlinoa some check issues:

```
> checking for unstated dependencies in examples ... OK
   WARNING
  'qpdf' is needed for checks on size reduction of PDFs

> checking package dependencies ... NOTE
  Imports includes 26 non-default packages.
  Importing from so many packages makes the package vulnerable to any of
  them becoming unavailable.  Move as many as possible to Suggests and
  use conditionally.

> checking dependencies in R code ... NOTE
  ':::' calls which should be '::':
    'automagic:::get_dependent_packages'
    'automagic:::get_package_details'
    See the note in ?`:::` about the use of this operator.
```

**Comment by jimbrig** [comment]
also, 

```
Updating polished documentation
Loading polished
Writing NAMESPACE
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\utils.R:103] @noRd has no parameters
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:73] undocumented R6 methods: `get_invite_by_email`, `find`, `sign_in_email`, `register_email`, `refresh_email_verification`, ...
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:73] undocumented R6 fields: `firebase_config`, `is_invite_required`, `sign_in_providers`, `is_email_verification_required`, `is_auth_required`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:90] argument `firebase_config` undocumented for R6 method `config()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:90] argument `admin_mode` undocumented for R6 method `config()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:90] argument `is_invite_required` undocumented for R6 method `config()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:90] argument `sign_in_providers` undocumented for R6 method `config()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:90] argument `is_email_verification_required` undocumented for R6 method `config()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:90] argument `is_auth_required` undocumented for R6 method `config()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:240] argument `email` undocumented for R6 method `get_invite_by_email()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:253] argument `hashed_cookie` undocumented for R6 method `find()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:253] argument `page` undocumented for R6 method `find()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:289] argument `email` undocumented for R6 method `sign_in_email()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:289] argument `password` undocumented for R6 method `sign_in_email()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:289] argument `hashed_cookie` undocumented for R6 method `sign_in_email()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:353] argument `email` undocumented for R6 method `register_email()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:353] argument `password` undocumented for R6 method `register_email()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:353] argument `hashed_cookie` undocumented for R6 method `register_email()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:383] argument `session_uid` undocumented for R6 method `refresh_email_verification()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:383] argument `firebase_token` undocumented for R6 method `refresh_email_verification()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:432] argument `session_uid` undocumented for R6 method `set_signed_in_as()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:432] argument `signed_in_as` undocumented for R6 method `set_signed_in_as()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:432] argument `user_uid` undocumented for R6 method `set_signed_in_as()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:454] argument `user_uid` undocumented for R6 method `get_signed_in_as_user()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:498] argument `session_uid` undocumented for R6 method `set_inactive()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:498] argument `user_uid` undocumented for R6 method `set_inactive()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:519] argument `hashed_cookie` undocumented for R6 method `sign_out()`
Warning: [C:\Users\Jimmy Briggs\Dev\tychobra\polished\R\Sessions.R:519] argument `session_uid` undocumented for R6 method `sign_out()`
Writing NAMESPACE
Warning: Topic 'Sessions': no parameters to inherit with @inheritParams
```

**Comment by jimbrig** [comment]
Latest check results: ✔

```
-- R CMD check results -------------------------------- polished 0.3.0.9000 ----
Duration: 1m 34s

> checking package dependencies ... NOTE
  Imports includes 26 non-default packages.
  Importing from so many packages makes the package vulnerable to any of
  them becoming unavailable.  Move as many as possible to Suggests and
  use conditionally.

> checking dependencies in R code ... NOTE
  ':::' calls which should be '::':
    'automagic:::get_dependent_packages'
    'automagic:::get_package_details'
    See the note in ?`:::` about the use of this operator.

0 errors v | 0 warnings v | 2 notes x

R CMD check succeeded
```

File: DESCRIPTION [hunk 1]
```diff
@@ -1,10 +1,11 @@
 Package: polished
 Type: Package
 Title: Authentication and Administration for 'shiny' Apps
-Version: 0.2.0.9000
+Version: 0.3.0.9000
```
Comment (merlinoa) [inline DESCRIPTION:4]: we don't want to bump to version 0.3.0 until the next CRAN release.  If you want to bump the dev version, the next version should be 0.2.0.9001.  Please update.

File: R/package_deps.R [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  # return if no detections
+  if (length(init_pkg_names) == 0) {
+    cli::cli_alert_warning("warning: no packages found in specified directory")
+    return(invisible(NULL))
+  }
+
+  # validate packages
+  hold <- lapply(init_pkg_names,
+                 purrr::safely(automagic::get_package_details, quiet = verbose))
```
Comment (merlinoa) [inline R/package_deps.R:54]: should it be quiet = !verbose.  Or you could change the argument name to "quiet".  Up to you

Comment (jimbrig) [reply R/package_deps.R]: I made this always quiet

**Pull request merged**